librz/asm/p/asm_chip8.c

```diff
@@ -11,82 +11,81 @@ static int chip8_disassemble(RzAsm *a, RzAsmOp *op, const ut8 *b, int l) {
+		case 0x7: rz_asm_op_setf_asm(op, "subn v%1x, v%1x", x, y); break;
+		case 0xE: rz_asm_op_setf_asm(op, "shl v%1x, v%1x", x, y); break;
 		}
 	} break;
-	case 0x9000: buf_asm = rz_str_newf("sne v%1x, v%1x", x, y); break;
-	case 0xA000: buf_asm = rz_str_newf("ld i, 0x%03x", nnn); break;
-	case 0xB000: buf_asm = rz_str_newf("jp v0, 0x%03x", nnn); break;
-	case 0xC000: buf_asm = rz_str_newf("rnd v%1x, 0x%02x", x, kk); break;
-	case 0xD000: buf_asm = rz_str_newf("drw v%1x, v%1x, 0x%01x", x, y, nibble); break;
+	case 0x9000: rz_asm_op_setf_asm(op, "sne v%1x, v%1x", x, y); break;
+	case 0xA000: rz_asm_op_setf_asm(op, "ld i, 0x%03x", nnn); break;
+	case 0xB000: rz_asm_op_setf_asm(op, "jp v0, 0x%03x", nnn); break;
+	case 0xC000: rz_asm_op_setf_asm(op, "rnd v%1x, 0x%02x", x, kk); break;
+	case 0xD000: rz_asm_op_setf_asm(op, "drw v%1x, v%1x, 0x%01x", x, y, nibble); break;
 	case 0xE000: {
 		if (kk == 0x9E) {
-			buf_asm = rz_str_newf("skp v%1x", x);
+			rz_asm_op_setf_asm(op, "skp v%1x", x);
 		} else if (kk == 0xA1) {
-			buf_asm = rz_str_newf("sknp v%1x", x);
+			rz_asm_op_setf_asm(op, "sknp v%1x", x);
 		}
 	} break;
 	case 0xF000: {
 		switch (kk) {
-		case 0x07: buf_asm = rz_str_newf("ld v%1x, dt", x); break;
-		case 0x0A: buf_asm = rz_str_newf("ld v%1x, k", x); break;
-		case 0x15: buf_asm = rz_str_newf("ld dt, v%1x", x); break;
-		case 0x18: buf_asm = rz_str_newf("ld st, v%1x", x); break;
-		case 0x1E: buf_asm = rz_str_newf("add i, v%1x", x); break;
-		case 0x29: buf_asm = rz_str_newf("ld f, v%1x", x); break;
-		case 0x33: buf_asm = rz_str_newf("ld b, v%1x", x); break;
-		case 0x55: buf_asm = rz_str_newf("ld [i], v%1x", x); break;
-		case 0x65: buf_asm = rz_str_newf("ld v%1x, [i]", x); break;
-		case 0x30: buf_asm = rz_str_newf("ld hf, v%1x", x); break;
-		case 0x75: buf_asm = rz_str_newf("ld r, v%1x", x); break;
-		case 0x85: buf_asm = rz_str_newf("ld v%1x, r", x); break;
+		case 0x07: rz_asm_op_setf_asm(op, "ld v%1x, dt", x); break;
+		case 0x0A: rz_asm_op_setf_asm(op, "ld v%1x, k", x); break;
+		case 0x15: rz_asm_op_setf_asm(op, "ld dt, v%1x", x); break;
+		case 0x18: rz_asm_op_setf_asm(op, "ld st, v%1x", x); break;
+		case 0x1E: rz_asm_op_setf_asm(op, "add i, v%1x", x); break;
+		case 0x29: rz_asm_op_setf_asm(op, "ld f, v%1x", x); break;
+		case 0x33: rz_asm_op_setf_asm(op, "ld b, v%1x", x); break;
+		case 0x55: rz_asm_op_setf_asm(op, "ld [i], v%1x", x); break;
+		case 0x65: rz_asm_op_setf_asm(op, "ld v%1x, [i]", x); break;
+		case 0x30: rz_asm_op_setf_asm(op, "ld hf, v%1x", x); break;
+		case 0x75: rz_asm_op_setf_asm(op, "ld r, v%1x", x); break;
+		case 0x85: rz_asm_op_setf_asm(op, "ld v%1x, r", x); break;
 		}
 	} break;
+	default: rz_asm_op_set_asm(op, "invalid"); break;
 	}
-	if (!buf_asm) {
-		rz_strbuf_set(&op->buf_asm, "invalid");
-	} else {
-		rz_strbuf_set(&op->buf_asm, buf_asm);
-		free(buf_asm);
+
+	if (!*rz_strbuf_get(&op->buf_asm)) {
+		rz_asm_op_set_asm(op, "invalid");
```

Comment:
i don't think this will be needed if all the cases have a `invalid` default case.

Reply:
or we could set `invalid` at the beginning and allow to change it again when it gets overwritten by the next set_asm